src/main/java/com/lambdatest/tunnel/Tunnel.java

```diff
@@ -584,58 +553,56 @@ public void runCommand(String command) throws IOException {
 
     public void runCommandV2(String[] command) throws IOException {
         try {
-            // ProcessBuilder processBuilder = new ProcessBuilder(command);
             Runtime run = Runtime.getRuntime();
             process = run.exec(command);
             Boolean update = false;
-            long start = System.currentTimeMillis();
-            long end = start;
+            waitForTunnelToStart();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void waitForTunnelToStart() throws IOException {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(HTTP_TIMEOUT)
+                .setSocketTimeout(HTTP_TIMEOUT)
+                .setConnectionRequestTimeout(HTTP_TIMEOUT)
+                .build();
+        CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
 
-            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
-            String line = null;
+        for (int retryCount = 0; retryCount < MAX_TUNNEL_STARTUP_RETRY_COUNT; retryCount++) {
             try {
-                while ((line = reader.readLine()) != null) {
-                    if (line.contains("Downloading update")) {
-                        update = true;
-                    } else {
-                        if (update) {
-                            System.out.println("Tunnel is updated. restarting...");
-                            runCommandV2(command);
-                        }
-                        try {
-                            if (t1.port != null) {
-                                BufferedReader br = new BufferedReader(
-                                        new FileReader(String.valueOf(".lambdatest/tunnelprocs/" + t1.port) + ".txt"));
-                                if (br.readLine() != null) {
-                                    tunnelFlag = true;
-                                    System.out.println("Tunnel Started Successfully");
-                                    break;
-                                }
-                            }
-                        } catch (Exception e) {
-                            // System.out.println("Not found any file");
+                Thread.sleep(TUNNEL_INITIAL_WAIT_AND_RETRY_BACKOFF);

```

Comment:
If thread.sleep() gets interrupted, due to the usage of the try-catch block it will log the exception and it will proceed to the next iteration.
Is this the expected flow or do have to stop retrying ?

Reply:
Yes. expected flow. It should retry.